src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSourceContext.java

```diff
@@ -85,6 +86,16 @@
      * {@code true} if notifications should be disabled in this context.
      */
     private boolean notificationsDisabled;
+    /**
+     * The {@link PullRequestNamingStrategy} to use in this context.
+     */
+    @NonNull
+    private PullRequestNamingStrategy pullRequestNamingStrategy = PullRequestNamingStrategy.PR_ID;
+    /**
+     * The pull request naming strategy exclude {@link Pattern} to use in this context.
+     */
+    @NonNull
+    private Pattern pullRequestNamingExcludePattern = Pattern.compile("");
```

Comment:
What about setting this only if the Getter is called and `pullRequestNamingExcludePattern` is null?
This is might be especially useful when loading from configurations, as the pattern is not compiled for every SCM config loaded.

Reply:
Done (please check that it is to your liking).